### DISCUSSION TITLE
Passing a big amount of data to shaders

### DISCUSSION THREAD

**Discussion by koalex76** [visgl/deck.gl#8824] [Q&A]
Hello. I have my custom extension with a custom attribute added via attributeManager and I need to pass a dozen of scalar parameters from the accessor into the vertex shader.  How do I do it?  What data structure should I use?

**Comment by ibgreen** [comment]
Each attribute is limited to four four byte values (vec4, ivec4, uvec4) (i.e. 16 bytes) and you can only have 16 attributes in total, a limit that is easy to hit since your layer is already using several attributes

If your scalar parameters are small you can bit pack them into the attribute values. That is easier if you are on GLSL 3.00 (which is enforced by deck.gl v9). You'd need to look up GLSL docs to see how to do that.

**Reply by koalex76** [reply]
Thank you. 

Maybe it's possible to use textures for that?
